src/App.tsx

```diff
@@ -15,48 +14,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface State {
+  sortedGoods: string[];
+  showResetButton: boolean;
+  sortByAlphabeticallyActive?: boolean;
+  sortByLengthActive?: boolean;
+  reverseActive?: boolean;
+}
+
 export const App: React.FC = () => {
```

Comment:
The task required the use of an enum for sorting options, however, no enums are being used for sort types. Consider replacing the string literals ('alphabetical', 'length', 'reverse') with an enum for better type safety.